src/Tabs/Tabs.js

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import Flex from '../Flex';
+import Tab from './Tab';
+import { ENTER_KEY, SPACEBAR } from '../constants';
+
+class Tabs extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const { activeTabLabel, children } = this.props;
+
+    this.state = {
+      activeTabLabel: activeTabLabel || children[0].props.label
+    };
+  }
+
+  get activeTabContent() {
+    const { activeTabLabel } = this.state;
+    const { children } = this.props;
+    const activeTab = children.find(
+      ({ props: { label } }) => label === activeTabLabel
+    );
+
+    return activeTab.props.content;
+  }
+
+  activateTab = tab => {
+    const { label } = tab.props;
+
+    this.setState({ activeTabLabel: label });
+  };
+
+  handleKeyPress = (key, tab) => {
+    if (key === ENTER_KEY || key === SPACEBAR) {
+      this.activateTab(tab);
+    }
+  };
+
+  render() {
+    const { activeTabLabel } = this.state;
+    const { Component, children, ...props } = this.props;
+
+    return (
+      <>
+        <Component {...props}>
+          {children.map(tab => {
+            const {
+              label,
+              children: tabChildren,
+              content,
+              ...tabProps
+            } = tab.props;
+            const active = activeTabLabel === label;
+
+            return (
+              <Tab
```

Comment:
Is the intention that you can pass any kind of component as children or specifically use `Tab` components like the storybook story shows?

If the expectation is to use `Tab` perhaps we should use `React.cloneElement` here instead of using the JSX `Tab` ourself here? It's a common pattern when aiming to manipulate props for children that are passed in.

https://frontarm.com/james-k-nelson/passing-data-props-children/#cloning-children

Reply:
Ah, good call